src/runtime_src/core/tools/xbmgmt2/ReportPlatform.cpp

```diff
@@ -86,20 +94,19 @@ ReportPlatform::getPropertyTree20201( const xrt_core::device * _pDevice,
   _pt.put("platform.installed_shell.vbnv", installedDSA.front().name);
   _pt.put("platform.installed_shell.sc_version", installedDSA.front().bmcVer);
   _pt.put("platform.installed_shell.id", (boost::format("0x%x") % installedDSA.front().timestamp));
-  _pt.put("platform.shell_upto_date", true);
+  _pt.put("platform.installed_shell.file", installedDSA.front().file);
+  _pt.put("platform.shell_upto_date", same_shell( on_board_rom_info.get<std::string>("vbnv"), on_board_rom_info.get<std::string>("id"), installedDSA.front()));

```

Comment:
If "vbnv" or "id" doesn't exist, the boost property tree get() method will throw an exception. 

Most likely we don't wish to do that (especially since this code will possible be used for edge devices).   In this case, it might be best to give the get() method a default value.  For example:
`on_board_rom_info.get<std::string>("vbnv","")`

Same below for "sc_version" and "platform.sc_upto_date"

Reply:
That's a good idea. I missed that. Thanks!